rewards/src/tests.rs

```diff
@@ -593,3 +593,35 @@ fn transfer_share_and_rewards() {
 		);
 	});
 }
+
+#[test]
+fn transfer_share_and_rewards_self_transfer() {
+	ExtBuilder::default().build().execute_with(|| {
+		RewardsModule::add_share(&ALICE, &DOT_POOL, 100);
+		assert_ok!(RewardsModule::accumulate_reward(&DOT_POOL, NATIVE_COIN, 100));
+
+		assert_noop!(
+			RewardsModule::transfer_share_and_rewards(&BOB, &DOT_POOL, 50, &BOB),
+			Error::<Runtime>::ShareDoesNotExist
+		);
+		assert_noop!(
+			RewardsModule::transfer_share_and_rewards(&ALICE, &DOT_POOL, 200, &ALICE),
+			Error::<Runtime>::CanSplitOnlyLessThanShare
+		);
+
+		let pool_info = RewardsModule::pool_infos(DOT_POOL);
+		assert_ok!(RewardsModule::transfer_share_and_rewards(&ALICE, &DOT_POOL, 33, &ALICE));
+
+		let new_pool_info = RewardsModule::pool_infos(DOT_POOL);
+		assert_eq!(pool_info, new_pool_info, "reward transfer does not affect the pool");
```

Comment:
Should it be transferred to BOB to verify this case?

Reply:
this is to test transfer to self, which previously wasn't handled correctly